src/v/model/record.h

```diff
@@ -858,18 +858,16 @@ class record_batch
 template<typename Func>
 inline ss::future<>
 for_each_record(const model::record_batch& batch, Func&& f) {
```

Comment:
nit: this commit and https://github.com/redpanda-data/redpanda/pull/13297/commits/008943922fd906d0af1693dd932407f75968ad78 and https://github.com/redpanda-data/redpanda/pull/13297/commits/176554c2742033a85e3b3312f70dca56f24ce1ca

Reply:
It could be? I'm not sure the guidelines on when to combine commits or not, so I'm trying to keep them all as isolated as possible.

Reply:
Typically, each commit introduces a single logical change. These commits seem to share the same logic.

Reply:
They are similar functions, but are not the same: some are async, some take l value references, etc. They where not introduced in the same commit either, so I don't really feel like it's really better to combine the commits here.